crypto/ec/ec_key.c

```diff
@@ -1112,17 +1115,36 @@ static int ecdsa_keygen_pairwise_test(EC_KEY *eckey, OSSL_CALLBACK *cb,
     OSSL_SELF_TEST_onbegin(st, OSSL_SELF_TEST_TYPE_PCT,
                            OSSL_SELF_TEST_DESC_PCT_ECDSA);
 
+    if (ctx == NULL
```

Comment:
I am fairly opposed to what you have done here.. It makes no difference to the underlying operation that is performed regardless of whether the message is digested or not.

The long winded way of doing this properly can be seen in the self_test_kats.c file..

i.e using EVP_PKEY_CTX_new_from_pkey(), EVP_SIGNATURE_fetch(), EVP_PKEY_sign_message_init() and EVP_PKEY_sign().

These API's are all exposed/duplicated inside the fips provider for the sole purpose of the self tests.

Reply:
Yes, using EVP_PKEY would be much better, like it is done in kats.

How can I create PKEY from a deprecated EC_KEY?

Reply:
I dont think we should do this (I dont like the idea of exposing the assign)
You could use 'from data' to create a EVP_PKEY but, 
I really think we should just leave this alone, since at the base level ECDSA_sign() is all OpenSSL provides.
It makes absolutely no difference whether you digest a message or not to whether a PCT is going to pass or fail, since we are using a constant message for the test - so this is really not logical.